app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingPresenterImpl.java

```diff
@@ -33,7 +34,7 @@
     public void setView(MoviesListingView view) {
         this.view = view;
         if(!showingSearchResult){
-            displayMovies();
+            //displayMovies();
```

Comment:
Why is this commented?

Reply:
The setView was also resulting in the displaying of fetched movies  after initializing the view.This resulted in multiple loading and displaying of initial page at first time. And overriding the restored state in other cases. Hence I have commented the display function here and managed the same through the firstPage() int the onViewCreated  for the initial time with out showing twice .And the second problem also solved.